src/app_service/service/settings/service.nim

```diff
@@ -966,3 +967,12 @@ QtObject:
       data.error = e.msg
       error "error saving social links", errDescription=data.error
     self.storeSocialLinksAndNotify(data)
+
+  proc isIncludeWatchOnlyAccount*(self: Service): bool =
+    return self.settings.includeWatchOnlyAccount
+
+  proc toggleIncludeWatchOnlyAccount*(self: Service) =
+    let newValue = not self.settings.includeWatchOnlyAccount
+    if(self.saveSetting(INCLUDE_WATCH_ONLY_ACCOUNT, newValue)):
+      self.settings.includeWatchOnlyAccount = newValue
+    self.events.emit(SIGNAL_INCLUDE_WATCH_ONLY_ACCOUNTS_UPDATED, Args())
```

Comment:
If you do not plan to emit any info about the error in case of failed saving to settings, then this line should go under `then` block of `if` condition.